pkg/scheduler/internal/cache/cache.go

```diff
@@ -240,8 +240,12 @@ func (cache *cacheImpl) UpdateSnapshot(logger klog.Logger, nodeSnapshot *Snapsho
 			if (len(existing.PodsWithAffinity) > 0) != (len(clone.PodsWithAffinity) > 0) {
 				updateNodesHavePodsWithAffinity = true
 			}
-			if (len(existing.PodsWithRequiredAntiAffinity) > 0) != (len(clone.PodsWithRequiredAntiAffinity) > 0) {
-				updateNodesHavePodsWithRequiredAntiAffinity = true
+			// We track nodes that have pods with anti-affinity, here we check if this node changed
+			// its status when we add or remove the pods with anti-affinity
+			if !updateNodesHavePodsWithRequiredAntiAffinity {
```

Comment:
I think this is not the root cause, because we check the `updateNodesHavePodsWithRequiredAntiAffinity` here to update node info at https://github.com/kubernetes/kubernetes/blob/f8ca582a06a8379968ddb7960adefeea3c467e36/pkg/scheduler/internal/cache/cache.go#L328-L330

So if we have 3 podsWithRequiredAntiAffinity before and now we have 4, it will not affect the `snapshot.havePodsWithRequiredAntiAffinityNodeInfoList`.